### PR TITLE
topology2: add sof-lnl-cs42l43-l0.tplg

### PR DESCRIPTION
**Pull request by bardliao**
This is for cs42l43 bridge/amp configuration.

### CODE REVIEW

**Review by charleskeepax — APPROVED** [review]
No hardware to test, but looks good to me.

File: tools/topology/topology2/production/tplg-targets-ace2.cmake [hunk 1]
```diff
@@ -17,6 +17,10 @@ SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack"
 SDW_AMP_FEEDBACK=false,SDW_SPK_STREAM=Playback-SmartAmp,SDW_DMIC_STREAM=Capture-SmartMic,\
 SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack"
 
+"cavs-sdw\;sof-lnl-cs42l43-l0\;PLATFORM=lnl,SDW_DMIC=1,NUM_HDMIS=3,NUM_SDW_AMP_LINKS=1,\
```
Comment (plbossart) [inline tools/topology/topology2/production/tplg-targets-ace2.cmake:20]: Do we need to set NUM_HDMIS=3?

I don't know why it's needed, isn't there a default already?
Also we need to clean-up the ones with NUM_HDMIS=0.

Comment (bardliao) [reply tools/topology/topology2/production/tplg-targets-ace2.cmake]: > Do we need to set NUM_HDMIS=3?
> 
> I don't know why it's needed, isn't there a default already? Also we need to clean-up the ones with NUM_HDMIS=0.

Yes, 3 is the default value of NUM_HDMIS, removed. Do you mean remove all NUM_HDMIS=0 for LNL like we did for MTL https://github.com/thesofproject/sof/pull/8789?

Comment (plbossart) [reply tools/topology/topology2/production/tplg-targets-ace2.cmake]: yes, the NUM_HDMIS was a work-around when HDMI was missing upstream patches, but I don't think it's still the case on LNL, is it?
@kv2019i I've obviously lost the plot here.

**Review by plbossart — APPROVED** [review]

**Comment by bardliao** [comment]
> Sorry @bardliao this fell through, are you Ok to rebase if still needed. Btw, needed of v2.10 ?

Sure, I will rebase it. It is needed for v2.10, but we need to wait for a DVT1 device to test.

**Comment by bardliao** [comment]
change: add `SDW_AMP_FMT_24=true,SDW_JACK_FMT_24=true`

**Comment by kv2019i** [comment]
@bardliao Any update, shall we move to v2.12?

**Comment by bardliao** [comment]
No, I think the topology is good.

**Pull request closed without merging**